Tests/Functional/Asynchronous/Expectations/main.swift

```diff
@@ -603,15 +633,20 @@ class ExpectationsTestCase: XCTestCase {
             ("test_expectationCreationOnSecondaryThread", test_expectationCreationOnSecondaryThread),
             ("test_expectationCreationWhileWaiting", test_expectationCreationWhileWaiting),
             ("test_runLoopInsideDispatch", test_runLoopInsideDispatch),
+
+            // waitForExpectations() + @MainActor
+            ("test_waitForExpectationsAsync", asyncTest(test_waitForExpectationsAsync)),
+            ("test_waitForExpectationsFromMainActor", asyncTest { test_waitForExpectationsFromMainActor($0) }),
```

Comment:
I wouldn't expect this one to use the `asyncTest` wrapper, since that means the wrapped test function will be invoked as if it were `async` and do the XCTestExpectation+XCTWaiter dance. Does plain
```suggestion
            ("test_waitForExpectationsFromMainActor", test_waitForExpectationsFromMainActor),
```
not work? If it doesn't, that feels like something else could be wrong

Reply:
It's causing a compiler crash which I've filed internally as a radar.

Reply:
rdar://111340003